components/esm/esmcommon.hpp

```diff
@@ -103,14 +107,27 @@ struct FIXED_STRING<4> : public FIXED_STRING_BASE<FIXED_STRING, 4>
     bool operator==(uint32_t v) const { return v == intval; }
     bool operator!=(uint32_t v) const { return v != intval; }
 
+    void assign(const std::string& value)
+    {
+        intval = 0;
+        size_t length = value.size();
+        if (length > 0)
+            data[0] = value[0];
+        if (length > 1)
+            data[1] = value[1];
+        if (length > 2)
+            data[2] = value[2];
+        if (length > 3)
```

Comment:
Doing 4 checks for strings of any length here seems a bit redundant.

Reply:
I added early-outs, but it is not a performance-critical function, and it is used just a couple of times in unit tests.